flattened options, fixed podman volume issues and added gopath volume

jgero opened the pull request:
Hi, it's the podman guy again :smile: 

My gopls language server started to crash after working with multiple instances of nvim open at the same time. I think it always happens when a second nvim instance creates a container that uses the exact same volume as another containers. It has something to do with how podman labels volumes. Adding the `:z` option at the end of the volume fixed it for me.

While already at it I decided to have a look at #14 and mounting another volume for the gopath and providing the environment variable. I took advantage of the cmd builder feature I added a while ago to modify how the gopls command is built.

I also changed how the custom image and cmd builder options have to be provided. I don't think there is a reason why this is behind another `additional_langauges` property, so I removed this property and flattened the options.

erkrnt commented:
All good! I appreciate the updates! If you don't mind please update the `README.md` to reflect any changes you've made that would require updates to configurations. I'm going to try this out later today.

jgero commented:
Okay I updated the example in the `README.md` with a new example that shows how the image and cmd can be used. I also added a very small paragraph of how to make it work with podman.

Because #36 is already merged I updated my PR and also changed the `GOPATH` volume to be mounted readonly as well.

In regards to testing: I am using the version of this PR of the plugin for Go programming since a few weeks with snippets and formatting set up, so the podman side of things works fine. Unfortunately I don't have the ability to test it with docker right now. But since podman an docker basically use the same command and the readonly volume thing seems to work it should be fine.

erkrnt commented:
@WhyNotHugo I tried but it looks like you can only retry after so long and the option was no longer available.

@jgero Thank you for resolving the conflicts and updates!
